pyccel/parser/semantic.py

```diff
@@ -43,56 +33,38 @@
 from pyccel.ast import Print
 from pyccel.ast import SymbolicPrint
 from pyccel.ast import Del
-from pyccel.ast import Assert
-from pyccel.ast import Comment, EmptyLine, NewLine
-from pyccel.ast import Break, Continue
+from pyccel.ast import EmptyLine
 from pyccel.ast import Slice, IndexedVariable, IndexedElement
 from pyccel.ast import FunctionHeader, ClassHeader, MethodHeader
-from pyccel.ast import VariableHeader, InterfaceHeader
-from pyccel.ast import MetaVariable
 from pyccel.ast import MacroFunction, MacroVariable
 from pyccel.ast import Concatenate
 from pyccel.ast import ValuedVariable
-from pyccel.ast import Argument, ValuedArgument
+from pyccel.ast import ValuedArgument
 from pyccel.ast import Is, IsNot
-from pyccel.ast import Import, TupleImport
+from pyccel.ast import Import
 from pyccel.ast import AsName
-from pyccel.ast import AnnotatedComment, CommentBlock
 from pyccel.ast import With, Block
 from pyccel.ast import List, Dlist, Len
 from pyccel.ast import builtin_function as pyccel_builtin_function
 from pyccel.ast import builtin_import as pyccel_builtin_import
 from pyccel.ast import builtin_import_registery as pyccel_builtin_import_registery
-from pyccel.ast import Macro
-from pyccel.ast import MacroShape
-from pyccel.ast import construct_macro
-from pyccel.ast import SumFunction, Subroutine
-from pyccel.ast import Zeros, Where, Linspace, Diag, Complex, EmptyLike
+from pyccel.ast import Subroutine
+from pyccel.ast import Zeros, Where, Linspace, Diag, EmptyLike
 from pyccel.ast import StarredArguments
 from pyccel.ast import inline, subs, create_variable, extract_subexpressions
 from pyccel.ast.core import get_assigned_symbols
 
-from pyccel.ast.core      import local_sympify, int2float, Pow, _atomic
+from pyccel.ast.core      import Pow, Add, Mul, And, Or, _atomic
+from pyccel.ast.core      import Eq, Ne, Lt, Le, Gt, Ge
+from pyccel.ast.core      import BooleanTrue, BooleanFalse
```

Comment:
This line is new but it is masked by
```
from sympy.logic.boolalg       import BooleanTrue, BooleanFalse
```
Before it is used

Reply:
So are `BooleanTrue` and `BooleanFalse` defined in pyccel or sympy or both?

Reply:
Both (see line 219 of ast/core.py)

Reply:
As the objective of this PR is to replace native Sympy objects with Pyccel AST nodes, I suppose that we should try to reduce the number of modules that import Sympy. Here I would like `pyccel.parser.semantic` not to use Sympy at all.

@saidctb @EmilyBourne Do you agree?

Reply:
done